### ISSUE TITLE
Incorrect query results

### ISSUE THREAD

**Issue by clibu** [appy-one/acebase#15]
I have a db with the following content:
````
songs: {
  klpmmym80000ggkkuzd3dtg1: {
    title: 'life on the Edge',
    artist: 'John Mayer',
    year: 2014,
    genre: 'pop',
    sex: 'none'
  },
  klpmqzur0000vgkkpn0g4uho: {
    title: 'Daughters',
    artist: 'John Mayer',
    year: 2014,
    genre: 'pop'
  },
  klpmtqrx0000xokkvim8n26w: {
    title: 'Crazy',
    artist: 'Gnarls Barkley',
    year: 2012,
    genre: 'rock'
  },
  klppvq780000hkkkvwdt4saf: {
    title: 'Lovely Sky',
    artist: 'Martha Smith',
    year: 2012,
    genre: 'folk'
  },
  klppvq7a0001hkkkgc27oea1: {
    title: 'Skylark',
    artist: 'Macey Grey',
    year: 2014,
    genre: 'pop'
  }
}
````

And these indexes:

````
db.indexes.create('songs', 'year')
db.indexes.create('songs', 'genre')
db.indexes.create('songs', 'year', { include: ['genre'] })
````

When I run this query:
````
db.query( 'songs' )
        .filter( 'year', '>=', 2012 )
        .filter( 'genre', 'in', ['jazz','rock','blues'])
        .sort( 'title' )
        .get()
````
I get 5 songs:
````
snapshot.val: { title: 'Crazy', artist: 'Gnarls Barkley', year: 2012, genre: 'rock' }
snapshot.val: { title: 'Daughters', artist: 'John Mayer', year: 2014, genre: 'pop' }
snapshot.val: { title: 'Lovely Sky',  artist: 'Martha Smith',  year: 2012,  genre: 'folk'}
snapshot.val: { title: 'Skylark', artist: 'Macey Grey', year: 2014, genre: 'pop' }
snapshot.val: {  title: 'life on the Edge',  artist: 'John Mayer',  year: 2014,  genre: 'pop',  sex: 'none'}
````

When only the first one should match.  If I do this without the last _(composite)_ index I get the correct results.

It may well be that this combination of indexes is invalid or doesn't make sense!

**Comment by appy-one** [comment]
It should only use the last index for that query because it has both year and genre. I'll have a look what's going on there.

**Comment by appy-one** [comment]
Fixed published in v1.1.2

**Comment by clibu** [comment]
@appy-one All good now. Thanks.
